AdsLib/Router.h

```diff
@@ -26,6 +26,8 @@
 #include "AdsDef.h"
 
 struct Router {
+	virtual ~Router() {};

```

Comment:
I don't think there is a need to explicitly add a destructor here.

Reply:
Missing destructor causes warnings in Eclipse, but I will remove it.